src/index.ts

```diff
@@ -67,65 +63,84 @@ export function generateWords(nr: number, words: string[] = []): string {
     nr = Math.abs(nr);
   }
 
+  let remainder = 0;
+  let quotient: number;
+  let integer: number;
+  let decimals: number;
+
   switch (true) {
     case (nr < 20):
-      remainder = 0;
-      word = LESS_THAN_TWENTY[Math.trunc(nr)];
-      word += parseDecimals(nr);
+      [,, integer, decimals] = parseNumber(nr);
```

Comment:
parseNumber(nr), the name is quite generic considering that this method returns and array of numbers, of which you should be very aware who they are and what they are for considering the position in the array.  
The function itself seems to breach the principle "Single responsibility principle".
I would support an oop approach, where you use a typed object with a proper interface defined, this way you will be aware all the time what each parsed number stands for.